recipes/libgeotiff/all/conanfile.py

```diff
@@ -66,39 +69,59 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "cmake"))
         tools.rmdir(os.path.join(self.package_folder, "doc"))
         tools.rmdir(os.path.join(self.package_folder, "share"))
+        self._create_cmake_module_variables(
+            os.path.join(self.package_folder, self._module_subfolder, self._module_vars_file)
+        )
         self._create_cmake_module_alias_targets(
-            os.path.join(self.package_folder, self._module_subfolder, self._module_file),
-            {
-                "geotiff_library": "GeoTIFF::GeoTIFF",
-                "geotiff_library": "geotiff::geotiff"
-            }
+            os.path.join(self.package_folder, self._module_subfolder, self._module_target_file),
+            {"geotiff_library": "geotiff::geotiff"}
         )
 
+    @staticmethod
+    def _create_cmake_module_variables(module_file):
+        content = textwrap.dedent("""\
+            if(DEFINED GeoTIFF_FOUND)
+                set(GEOTIFF_FOUND ${GeoTIFF_FOUND})
+            endif()
+            if(DEFINED GeoTIFF_INCLUDE_DIR)
+                set(GEOTIFF_INCLUDE_DIR ${GeoTIFF_INCLUDE_DIR})
+            endif()
+            if(DEFINED GeoTIFF_LIBRARIES)
+                set(GEOTIFF_LIBRARIES ${GeoTIFF_LIBRARIES})
+            endif()
+        """)
+        tools.save(module_file, content)
+
     @staticmethod
     def _create_cmake_module_alias_targets(module_file, targets):
         content = ""
         for alias, aliased in targets.items():
-            content += (
-                "if(TARGET {aliased} AND NOT TARGET {alias})\n"
-                "    add_library({alias} INTERFACE IMPORTED)\n"
-                "    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})\n"
-                "endif()\n"
-            ).format(alias=alias, aliased=aliased)
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
         tools.save(module_file, content)
 
     @property
     def _module_subfolder(self):
         return os.path.join("lib", "cmake")
 
     @property
-    def _module_file(self):
+    def _module_vars_file(self):
+        return "conan-official-{}-variables.cmake".format(self.name)
+
+    @property
+    def _module_target_file(self):
         return "conan-official-{}-targets.cmake".format(self.name)
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "GeoTIFF"
         self.cpp_info.names["cmake_find_package_multi"] = "geotiff"
-        self.cpp_info.builddirs = [self._module_subfolder]
-        self.cpp_info.build_modules = [os.path.join(self._module_subfolder, self._module_file)]
+        self.cpp_info.builddirs.append(self._module_subfolder)
```

Comment:
Think aloud here... There must be a technical detail about how this works which I am not familiar with.

If I understand correctly the previous way by assigning just this subfolder meant it **only looked** in `package_folder/lib/cmake` for the `FindPackage.cmake` which causes things to break.

The default is `[""]` which Conan internal translates to the `package_folder` (🧙) 

❓ If we put nothing in that folder why does it matter? What is this settings overriding the causes the CMake to look in the wrong path(s)?

The generated files are are output into the build folder which I presume must be included?

***

The docs https://docs.conan.io/en/latest/reference/conanfile/attributes.html#cpp-info which do not help me understand

NAME | DESCRIPTION
-- | --
`self.cpp_info.builddirs` | Ordered list with build scripts directory paths. Defaulted to [""] (Package folder directory)CMake generators will search in these dirs for files like findXXX.cmake

Reply:
Explained above: https://github.com/conan-io/conan-center-index/pull/4707#issuecomment-787102357

Reply:
`builddirs` seems to have more responsability than the doc says, e.g setting `CMAKE_PREFIX_PATH` in `cmake` generator